tensorflow/python/compiler/tensorrt/trt_convert_test.py

```diff
@@ -929,13 +938,12 @@ def _InputFn():
         input_saved_model_dir, max_workspace_size_bytes=10 << 40)
     converter.convert()
 
-    os.environ["TF_TRT_ALLOW_ENGINE_NATIVE_SEGMENT_EXECUTION"] = "False"
     with self.assertRaisesRegex(
         errors.AbortedError,
         r"User disallowed engine native segment execution"):
-      converter.build(input_fn=_InputFn)
+      with trt_utils.disable_native_segment_fallback():
+        converter.build(input_fn=_InputFn)
 
-    os.environ["TF_TRT_ALLOW_ENGINE_NATIVE_SEGMENT_EXECUTION"] = "True"
     converter.build(input_fn=_InputFn)
```

Comment:
Will putting the assert and the env variable reset in a try-finally block like here work?

os.environ["TF_TRT_ALLOW_ENGINE_NATIVE_SEGMENT_EXECUTION"] = "False"
try:
    with self.assertRaisesRegex(
        errors.AbortedError,
        r"User disallowed engine native segment execution"):
      converter.build(input_fn=_InputFn)
finally:
    # Always reset the environment variable.
    os.environ["TF_TRT_ALLOW_ENGINE_NATIVE_SEGMENT_EXECUTION"] = "True"

# Continue the test by repeating the build method with native segments allowed. 
converter.build(input_fn=_InputFn)

Reply:
Absolutely it would work as discussed by email. I made the choice to implement this in a utils function since it could be useful for other tests/usecases: https://github.com/tensorflow/tensorflow/commit/da5eba6e9a#diff-64310e778a6bbfd7b6505d834ebce26a563f76ee821db8542227c329c76bbf84R52-R58

Let me know if that works for you

Reply:
What you have here is still not what I was talking about. We shouldn't need disable_native_segment_fallback and @contextmanager.

Reply:
contextmanager removed. Replace by a try/finally block in the unittest